src/plugins/visualizations/server/migrations/visualization_common_migrations.ts

```diff
@@ -199,3 +199,21 @@ export const commonRemoveMarkdownLessFromTSVB = (visState: any) => {
 
   return visState;
 };
+
+export const commonUpdatePieVisApi = (visState: any) => {
+  if (visState && visState.type === 'pie') {
+    const { addLegend, labels, ...restParams } = visState.params;
+    // eslint-disable-next-line @typescript-eslint/naming-convention
+    const { last_level, ...restLabels } = labels ?? {};
```

Comment:
I second that.

Reply:
@flash1293, @stratoula, addressed this issue at the current commit: https://github.com/elastic/kibana/pull/122420/commits/5f7ceafc89ebeb5cf1a77f46b640e96ded62f183

Reply:
@flash1293, we couldn't do that, because to_ast logic is related to the actual shape of the expression arguments. If we would override it, we'd lose the connection between vis_types and the expression. I've tried and come to the idea, it is better to leave it as it has been before.